libraries/stdlib/stdlib_defs.mtlx

```diff
@@ -44,7 +44,22 @@
   <typedef name="geomnamearray"/>
 
   <!-- ======================================================================== -->
-  <!-- Geometric Properties
+  <!-- Unit types                                                               -->
+  <!-- ======================================================================== -->
+   <unittypedef name="length" default="meter">
+      <unitdef name="nanometer" scale="1000000000.0" />
```

Comment:
Offset is never used so is not specified. The unit converter has been simplified as well.
If a conversion needs offset than that attribute + support can be added for that type.